exp/clusterresourceset/controllers/clusterresourceset_helpers.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	clusterresourcesetv1 "sigs.k8s.io/cluster-api/exp/clusterresourceset/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// ToUnstructured takes a YAML and converts it to a list of Unstructured objects
+func ToUnstructured(rawyaml []byte) ([]unstructured.Unstructured, error) {
```

Comment:
The is a similar function in clusterctl https://github.com/kubernetes-sigs/cluster-api/blob/c9e23102da2531e3d723fcb44401327654db869a/cmd/clusterctl/internal/util/yaml.go#L57-L94
Most probably, we should move this to a util

Reply:
Yes, I used that, let's move it in a follow up PR.

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/3133